docs/running-on-yarn.md

```diff
@@ -472,6 +472,36 @@ To use a custom metrics.properties for the application master and executors, upd
 </tr>
 </table>
 
+#### Available patterns for SHS custom executor log URL
+
+<table class="table">
```

Comment:
It would be great if here there were an example of how to get this to point to the JHS for app logs after the application finished.

Reply:
I just gave it a try but found one thing which blocks me pointing JHS URL. I have no idea where port `25454` in JHS URL came from.

* NM URL

```
http://<node-host>:8042/node/containerlogs/container_e01_1547723113049_0010_01_000001/spark/stdout?start=-4096
```

* JHS URL

```
http://<job-history-server-host>:19888/jobhistory/logs/<node-host>:25454/container_e01_1547723113049_0010_01_000001/container_e01_1547723113049_0010_01_000001/spark/stdout?start=-4096
```

Reply:
At least on my cluster, that port is the NM's IPC (not http) port; so it doesn't come from the Spark application.

Reply:
Oh OK. Maybe we could get the information from YarnConfiguration? If we think it's worth to provide it as attribute I'll apply the change.

Reply:
Hmm, it would be useful to have that if only to have a simple, "built in" use case for this. The SHS could point directly to the JHS and then NMs can be decomissioned without breaking the link.

YARN could record `NM_HTTP_ADDRESS` and `NM_IPC_ADDRESS`, for example. (Note the name change from your existing name.)